tensorflow/core/kernels/resize_nearest_neighbor_op.cc

```diff
@@ -152,21 +158,30 @@ struct ResizeNearestNeighbor<CPUDevice, T, half_pixel_centers, align_corners> {
         if (half_pixel_centers) {
           in_y = std::max(static_cast<Eigen::Index>(0), in_y);
         }
-        for (Eigen::Index x = 0; x < out_width; ++x) {
-          Eigen::Index in_x = std::min(
-              (align_corners)
-                  ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
-                  : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
-              in_width - 1);
-          if (half_pixel_centers) {
-            in_x = std::max(static_cast<Eigen::Index>(0), in_x);
-          }
-          std::copy_n(&input(b, in_y, in_x, 0), channels, &output(b, y, x, 0));
+        Eigen::Index in_x = std::min(
+            (align_corners)
+                ? static_cast<Eigen::Index>(roundf(scaler(x, width_scale)))
+                : static_cast<Eigen::Index>(floorf(scaler(x, width_scale))),
+            in_width - 1);
+        if (half_pixel_centers) {
+          in_x = std::max(static_cast<Eigen::Index>(0), in_x);
         }
+        std::copy_n(&input(bs, in_y, in_x, 0), channels, &output(bs, y, x, 0));
       }
-    }
+    };
+    Eigen::Index N = batch_size * out_height * out_width;
+    Shard(worker_threads.num_threads, worker_threads.workers, N, 1000.0,
+          ParallelResize);  // TODO: Come up with a good cost estimate:
+                            // 3500:26~27fps, 1000:27~28fps.
     return true;
   }
+  bool operator()(const CPUDevice& d, typename TTypes<T, 4>::ConstTensor input,
```

Comment:
Why do we need `ParallelExecute` to be another function? The code is not used anywhere else and can just be in here.

Reply:
Changes as requested.